engine/src/main/java/org/pentaho/di/base/AbstractBaseCommandExecutor.java

```diff
@@ -234,6 +237,31 @@ public boolean isEnabled( final String value ) {
     return YES.equalsIgnoreCase( value ) || Boolean.parseBoolean( value ); // both are NPE safe, both are case-insensitive
   }
 
+  /**
+   * Decodes the provided base64String into the specified filePath. Parent directories must already exist.
+   *
+   * @param base64String String BASE64 representation of a file
+   * @param filePath     String The path to which the base64String is to be decoded
+   * @param deleteOnExit boolean Delete created file on JVM exit if true
+   */
+  protected void decodeBase64StringToFile( String base64String, String filePath, boolean deleteOnExit ) {
+    //Decode base64String to byte[]
+    byte[] decodedBytes = Base64.getDecoder().decode( base64String );
+
+    //Set file to delete on exit if true
+    if ( deleteOnExit ) {
```

Comment:
if I pass `deleteOnExit` as `false` ... would it **not** create the java.io.File ?

Reply:
it will still create the file on disk, but it won't delete it. FileOutputStream didn't have a mechanism to cleanup the created file, so I used File in that case